packaging/test_rule_engine_plugin_metadata_guard.py

```diff
@@ -223,6 +223,49 @@ def test_plugin_honors_admin_only_config_option_when_rodsusers_manipulate_metada
         # Clean up.
         self.rods.assert_icommand(['imeta', 'rm', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
 
+    def test_plugin_disallows_rmw(self):
+        config = IrodsConfig()
+
+        # Set JSON configuration for the root collection.
+        root_coll = os.path.join('/', self.admin.zone_name)
+        json_config = json.dumps({
+            'prefixes': ['irods::'],
+            'admin_only': True
+        })
+        self.rods.assert_icommand(['imeta', 'set', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+        try:
+            with lib.file_backed_up(config.server_config_path):
+                self.enable_rule_engine_plugin(config)
+
+                # neither user should be able to use rmw
+                self.user.assert_icommand(['imeta', 'rmw', '-C', self.user.session_collection, '%', '%'], 'STDERR', ['SYS_NOT_ALLOWED'])
+                self.rods.assert_icommand(['imeta', 'rmw', '-C', self.rods.session_collection, '%', '%'], 'STDERR', ['SYS_NOT_ALLOWED'])
+        finally:
+            # Clean up.
+            self.rods.assert_icommand(['imeta', 'rm', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+    def test_plugin_cannot_be_bypassed_with_addw(self):
+        config = IrodsConfig()
+
+        # Set JSON configuration for the root collection.
+        root_coll = os.path.join('/', self.admin.zone_name)
+        json_config = json.dumps({
+            'prefixes': ['irods::'],
+            'admin_only': True
+        })
```

Comment:
We should cover the `editors` case as well.

Reply:
Since `addw` still works under normal logic, do I add a check here to ensure that `editors` *does* work? 
If `addw` is being deprecated, is there value in checking that it still works if we're hankering to get rid of it?

Reply:
Given it's deprecated, I think it's fine to ignore the `editors` case.